pantheon/src/main/java/tech/pegasys/pantheon/cli/PantheonCommand.java

```diff
@@ -202,62 +198,86 @@ public RpcApi convert(final String name) throws RpcApisConversionException {
-  //    names = {"--sync-mode"},
-  //    paramLabel = MANDATORY_MODE_FORMAT_HELP,
-  //    description =
-  //        "Synchronization mode (Value can be one of ${COMPLETION-CANDIDATES}, default:
-  // ${DEFAULT-VALUE})"
-  //  )
-  private final SyncMode syncMode = DEFAULT_SYNC_MODE;
+  @Option(
+    names = {"--network"},
+    paramLabel = MANDATORY_NETWORK_FORMAT_HELP,
+    description =
+        "Synchronize against the indicated network, possible values are ${COMPLETION-CANDIDATES}."
+            + " (default: ${DEFAULT-VALUE})"
+  )
+  private final NetworkName network = MAINNET;
 
+  /** @deprecated Deprecated in favour of --network option */
+  @Deprecated
   // Boolean option to indicate if the client have to sync against the ottoman test network
   // (see https://github.com/ethereum/EIPs/issues/650).
   @Option(
+    hidden = true,
     names = {"--ottoman"},
     description =
         "Synchronize against the Ottoman test network, only useful if using an iBFT genesis file"
             + " - see https://github.com/ethereum/EIPs/issues/650 (default: ${DEFAULT-VALUE})"
   )
   private final Boolean syncWithOttoman = false;
 
+  /** @deprecated Deprecated in favour of --network option */
+  @Deprecated
   @Option(
+    hidden = true,
     names = {"--rinkeby"},
     description =
         "Use the Rinkeby test network"
             + " - see https://github.com/ethereum/EIPs/issues/225 (default: ${DEFAULT-VALUE})"
   )
   private final Boolean rinkeby = false;
 
+  /** @deprecated Deprecated in favour of --network option */
+  @Deprecated
   @Option(
+    hidden = true,
     names = {"--ropsten"},
     description = "Use the Ropsten test network (default: ${DEFAULT-VALUE})"
   )
   private final Boolean ropsten = false;
 
+  /** @deprecated Deprecated in favour of --network option */
+  @Deprecated
   @Option(
+    hidden = true,
     names = {"--goerli"},
     description = "Use the Goerli test network (default: ${DEFAULT-VALUE})"
   )
   private final Boolean goerli = false;
 
   @Option(
-    names = {"--p2p-listen"},
-    paramLabel = MANDATORY_HOST_AND_PORT_FORMAT_HELP,
-    description = "Host and port for p2p peers discovery to listen on (default: ${DEFAULT-VALUE})",
+    names = {"--p2p-host"},
+    paramLabel = MANDATORY_HOST_FORMAT_HELP,
+    description = "Host for p2p peers discovery to listen on (default: ${DEFAULT-VALUE})",
     arity = "1"
   )
-  private final HostAndPort p2pHostAndPort = getDefaultHostAndPort(DefaultPeer.DEFAULT_PORT);
+  private final HostSpecifier p2pHost =
+      HostSpecifier.fromValid(autoDiscoverDefaultIP().getHostAddress());
 
   @Option(
+    names = {"--p2p-port"},
+    paramLabel = MANDATORY_PORT_FORMAT_HELP,
+    description = "Port for p2p peers discovery to listen on (default: ${DEFAULT-VALUE})",
+    arity = "1"
+  )
+  private final Integer p2pPort = DEFAULT_PORT;
+
+  /** @deprecated Deprecated in favour of --network option */
+  @Deprecated
+  @Option(
+    hidden = true,
```

Comment:
This probably shouldn't be hidden until we actually support setting the network ID in genesis config.  Otherwise there's no way to set a custom network ID.